packages/tools/kolibri-cli/src/migrate/shares/reuse.ts

```diff
@@ -0,0 +1,105 @@
+ * @returns {string[]} The found files
+ */
+export function filterFilesByExt(dir: string, ext: FileExtension | FileExtension[]): string[] {
+	ext = Array.isArray(ext) ? ext : [ext];
+	let files: string[] = [];
+	const dirPath = path.resolve(process.cwd(), dir);
+	fs.readdirSync(dirPath).forEach((file) => {
+		const fullPath = path.resolve(dir, file);
+		if (fs.lstatSync(fullPath).isDirectory()) {
+			files = files.concat(filterFilesByExt(fullPath, ext));
+		} else if (ext.includes(path.extname(fullPath).replace('.', '') as FileExtension)) {
+			files.push(fullPath);
+		}
+	});
+	return files;
+}
+
+/**
+ * This function is used to get the version of the package.json as string.
+ * @param {string} offsetPath The offset path to the package.json
+ * @returns {string} The package.json as string
+ */
+export function readPackageString(offsetPath: string): string {
+	offsetPath = path.resolve(offsetPath, 'package.json');
+	if (!fs.existsSync(offsetPath)) {
+		throw new Error(`${offsetPath} not found`);
+	}
+	return fs.readFileSync(offsetPath, 'utf8');
+}
+
+/**
+ * This function is used to get the version of the package.json.
+ * @param {string} offsetPath The offset path to the package.json
+ * @returns {PackageJson} The package.json as object
+ */
+export function readPackageJson(offsetPath: string): PackageJson {
+	const content = readPackageString(offsetPath);
+	let json: Record<string, unknown>;
+	try {
+		json = JSON.parse(content) as Record<string, unknown>;
+	} catch (err) {
+		throw new Error(`Error reading package.json: ${(err as Error).message}`);
+	}
+	return json as PackageJson;
+}
+
+/**
+ * This function is used to get the package manager install command.
+ * @param {string} baseDir The base directory to start searching for the package manager
+ * @returns {string} The package manager install command
+ */
+export function getPackageManagerInstallCommand(baseDir: string = process.cwd()) {
+	if (fs.existsSync(path.resolve(baseDir, 'pnpm-lock.yaml'))) return 'pnpm i';
+	if (fs.existsSync(path.resolve(baseDir, 'yarn.lock'))) return 'yarn';
+	if (fs.existsSync(path.resolve(baseDir, 'package-lock.json'))) return 'npm i';
+	baseDir = path.resolve(baseDir, '..');
+	if (baseDir === '/') {
+		throw new Error('Package manager could not detected.');
+	}
+	return getPackageManagerInstallCommand(baseDir);
+}
+
+export const isTagKebabCaseRegExp = /^kol-[a-z]+(-[a-z]+)*$/;
+export const isPropertyKebabCaseRegExp = /^(data-removed-)?_[a-z]+(-[a-z]+)*$/;
+
+/**
+ * Converts a kebab case string to a capital case string.
+ * @param {string} tag The kebab case string
+ * @returns {string} The capital case string
+ */
+export function kebabToCapitalCase(tag: string) {
+	return tag
+		.split('-') // Split on hyphen
+		.map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()) // Capitalize each word
+		.join(''); // Join without space
+}
+
+export const MODIFIED_FILES = new Set<string>();
+
+export type RemoveMode = 'comment' | 'delete';
```

Comment:
Ich finde, hier würde sich eine Lösung analog zur types.ts anbieten, bei der ein Array von Optionen definiert und der Type davon abgeleitet wird. Das Array könnte dann in `src/migrate/index.ts` wiederverwendet werden und dort die hardcoded `choices` ersetzen.